src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -111,7 +114,10 @@ def _extract_messages(self) -> None:
             assert isinstance(msg.value, BrokerValue)
             partition_offset = PartitionIdxOffset(msg.value.partition.index, msg.value.offset)
             try:
-                parsed_payload = json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
+                parsed_payload = self.__input_codec.decode(
```

Comment:
I think we need to run decode and validate in two separate steps. Otherwise it's not possible to continue on schema validation error. We have the same problem in snuba.